src/client/runner/automation/playback/dblclick.js

```diff
@@ -5,201 +5,206 @@ import { fromPoint as getElementFromPoint } from '../get-element';
-                    }
+        });
+    }
+
+    _secondClick () {
+        return this._mousedown()
+            .then(() => this._mouseup())
+            .then(() => this._click());
+    }
+
+    _mousedown () {
+        this.eventArgs = this._calculateEventArguments();
+
+        this._raiseTouchEvents();
+
+        return cursor
+            .leftButtonDown()
+            .then(() => {
+                var isBodyElement         = this.eventArgs.element.tagName.toLowerCase() === 'body';
+                var isContentEditable     = domUtils.isContentEditableElement(this.eventArgs.element);
+                var isContentEditableBody = isBodyElement && isContentEditable;
+                var activeElement         = domUtils.getActiveElement();
+
+                // NOTE: in IE, focus is not raised if the element was focused
+                // before the click, even if focus is lost during the mousedown
+                this.eventState.simulateFocus = !browserUtils.isIE || activeElement !== this.eventArgs.element ||
+                                                isContentEditableBody;
+
+                // NOTE: in WebKit and IE, the mousedown event opens the select element's dropdown.
+                // Therefore, we should prevent mousedown and hide the dropdown (B236416)
+                var needCloseSelectDropDown = (browserUtils.isWebKit || browserUtils.isIE) &&
+                                              domUtils.isSelectElement(this.element);
+
+                if (needCloseSelectDropDown)
+                    this._bindMousedownHandler();
+
+                this.eventState.simulateDefaultBehavior =
+                    eventSimulator.mousedown(this.eventArgs.element, this.eventArgs.options);
+
+                if (this.eventState.simulateDefaultBehavior === false) {
+                    this.eventState.simulateDefaultBehavior = needCloseSelectDropDown &&
+                                                              !this.eventState.mousedownPrevented;
                 }
+
+                return this._focus();
             });
-        },
+    }
+
+    _focus () {
+        //NOTE: we should not call focus after the second mousedown (except in IE) because of the native browser behavior
+        if (this.eventState.simulateDefaultBehavior === false || browserUtils.isIE)
+            return Promise.resolve();
+
+        // NOTE: If a target element is a contentEditable element, we need to call focusAndSetSelection directly for
+        // this element. Otherwise, if the element obtained by elementFromPoint is a child of the contentEditable
+        // element, a selection position may be calculated incorrectly (by using the caretPos option).
+        var elementForFocus = domUtils.isContentEditableElement(this.element) ?
+                              this.element : this.eventArgs.element;
+
+        return automationUtil.focusAndSetSelection(elementForFocus, this.eventState.simulateFocus, this.caretPos);
+    }
+
+    _mouseup () {
```

Comment:
`clickAutomation` with have the same `_focus, _mousedown, _mouseup, _click` methods, so it's code duplication. Let's use `clickAutomation` for the second click too